akka-stream/src/main/resources/reference.conf

```diff
@@ -72,6 +72,16 @@ akka {
         # of 1 on the corresponding dispatchers.
         fuzzing-mode = off
       }
+
+      io.tcp {
+        # The outgoing bytes are accumulated in a buffer while waiting for acknoledgment
+        # of pending write. This improves throughput for small messages (frames) without
+        # sacrificing latency. While waiting for the ack the stage will eagerly pull
+        # from upstream until the buffer exceeds this size. That means that the buffer may hold
+        # slightly more bytes than this limit (at most one element more). It can be set to 0
+        # to disable the usage of the buffer.
+        write-buffer-size = 16 KiB
```

Comment:
Googled around a bit but not sure if that's a good default, perhaps as good as any I guess, for a user-level buffer hm

Reply:
Bigger didn't increase throughput in `MaxThroughputSpec`, but smaller decreased it.

Reply:
On local box or jenkins / amazon?

Reply:
Basically, this adds a buffer before the asynchronous boundary between streams and Akka IO TCP implementation. It's pretty clear that this helps. How big the buffer should be depends on how fast the network is vs how small the chunks are and how much extra latency is added by the asynchronous boundary.

Reply:
In our tests buffer of 16KB-20KB was giving best performance as well, a little search on optimal size lead us to https://en.wikipedia.org/wiki/Jumbo_frame. Most systems we deal with have this jumbo frame setup and network stack is able to transfer 8K in one tcp packet and further search also suggest that there are some TCP stack optimizations if there is a queue of at least 2 packets.
I no longer cannot find this references and maybe it is not relevant here - but at least some explanation to magic number ;)